monai/transforms/utils.py

```diff
@@ -156,17 +156,26 @@ def rescale_array(
 ) -> NdarrayOrTensor:
     """
     Rescale the values of numpy array `arr` to be from `minv` to `maxv`.
+
+    Args:
+        arr: input array to rescale.
+        minv: minimum value of target rescaled array.
+        maxv: maxmum value of target rescaled array.
+        dtype: data type of output, if None, same as input array.
+
     """
-    if dtype is not None:
-        arr, *_ = convert_data_type(arr, dtype=dtype)
+    dtype_ = dtype or arr.dtype
+    arr, *_ = convert_data_type(arr, dtype=dtype_)
```

Comment:
Is this doing something that the original didn't do? If `dtype` is `None` then doing no conversion seems fine, converting when `type_` is `arr.dtype` seems like a no-op?

Reply:
Hi @ericspod ,

OK, sure, I changed the rescale_array back with more doc-strings for args, and updated `ScaleIntensity` to adjust the dtype in transform.
Could you please help take a look again?

Thanks in advance.